packages/eas-cli/src/build/utils/resourceClass.ts

```diff
@@ -0,0 +1,135 @@
+  [ResourceClass.MEDIUM]: BuildResourceClass.IosMedium,
+};
+
+const androidResourceClassToBuildResourceClassMapping: Record<
+  Exclude<
+    ResourceClass,
+    | ResourceClass.M1_EXPERIMENTAL
+    | ResourceClass.M1_MEDIUM
+    | ResourceClass.M1_LARGE
+    | ResourceClass.INTEL_MEDIUM
+  >,
+  BuildResourceClass
+> = {
+  [ResourceClass.DEFAULT]: BuildResourceClass.AndroidDefault,
+  [ResourceClass.LARGE]: BuildResourceClass.AndroidLarge,
+  [ResourceClass.MEDIUM]: BuildResourceClass.AndroidMedium,
+};
+
+async function resolveIosDefaultRequestedResourceClassAsync(
+  getDynamicProjectConfigAsync: DynamicConfigContextFn,
+  projectDir: string,
+  buildProfile: ProfileData<'build'>
+): Promise<ResourceClass> {
+  const { exp } = await getDynamicProjectConfigAsync({ env: buildProfile.profile.env });
+  const { sdkVersion } = exp;
+  const reactNativeVersion = await getReactNativeVersionAsync(projectDir);
+  if (
+    (sdkVersion && semver.satisfies(sdkVersion, '>=48')) ||
+    (reactNativeVersion && semver.satisfies(reactNativeVersion, '>=0.71.0'))
+  ) {
+    return ResourceClass.M1_MEDIUM;
+  } else {
+    return ResourceClass.DEFAULT;
+  }
+}
+
+function resolveAndroidResourceClass(
+  profileResourceClass?: ResourceClass,
+  resourceClassFlag?: ResourceClass
+): BuildResourceClass {
+  if (
+    resourceClassFlag &&
+    [
+      ResourceClass.M1_EXPERIMENTAL,
+      ResourceClass.M1_MEDIUM,
+      ResourceClass.M1_LARGE,
+      ResourceClass.INTEL_MEDIUM,
+    ].includes(resourceClassFlag)
+  ) {
+    throw new Error(`Resource class ${resourceClassFlag} is only available for iOS builds`);
+  }
+
+  const resourceClass = resourceClassFlag ?? profileResourceClass ?? ResourceClass.DEFAULT;
+
+  return androidResourceClassToBuildResourceClassMapping[
+    resourceClass as Exclude<
+      ResourceClass,
+      | ResourceClass.M1_EXPERIMENTAL
+      | ResourceClass.M1_MEDIUM
+      | ResourceClass.M1_LARGE
+      | ResourceClass.INTEL_MEDIUM
+    >
+  ];
+}
+
+async function resolveIosResourceClassAsync(
+  buildProfile: ProfileData<'build'>,
+  getDynamicProjectConfigAsync: DynamicConfigContextFn,
+  projectDir: string,
+  profileResourceClass?: ResourceClass,
+  resourceClassFlag?: ResourceClass
+): Promise<BuildResourceClass> {
+  const resourceClass =
+    resourceClassFlag ??
+    profileResourceClass ??
+    (await resolveIosDefaultRequestedResourceClassAsync(
+      getDynamicProjectConfigAsync,
+      projectDir,
+      buildProfile
+    ));
+
+  if (resourceClass === ResourceClass.M1_EXPERIMENTAL) {
+    Log.warn(`Resource class ${chalk.bold('m1-experimental')} is deprecated.`);
+  }
+
+  if ([ResourceClass.LARGE, ResourceClass.M1_LARGE].includes(resourceClass)) {
+    Log.warn(
+      `Large resource classes are not available for iOS builds yet. Your build will use the medium resource class.`
+    );
+  }
+
+  return iosResourceClassToBuildResourceClassMapping[resourceClass];
+}
+
+export async function resolveBuildResourceClassAsync(
```

Comment:
Nitpick: Place exported functions above non-exported. When you open a module, you're usually interested in the API, not implementation details.